src/main/java/com/hubspot/jinjava/interpret/InvalidArgumentException.java

```diff
@@ -0,0 +1,63 @@
+package com.hubspot.jinjava.interpret;
+
+import com.hubspot.jinjava.lib.Importable;
+
+public class InvalidArgumentException extends RuntimeException {
+
+  private final int lineNumber;
+  private final int startPosition;
+  private final String message;
+  private final String name;
+
+  public InvalidArgumentException(JinjavaInterpreter interpreter, Importable importable, InvalidReason invalidReason, int argumentNumber, Object... errorMessageArgs) {
+    this.message = String.format("Invalid argument in '%s': %s",
+        importable.getName(),
+        String.format("%s %s", formatArgumentNumber(argumentNumber + 1), String.format(invalidReason.getErrorMessage(), errorMessageArgs)));
+
+    this.lineNumber = interpreter.getLineNumber();
+    this.startPosition = interpreter.getPosition();
+    this.name = importable.getName();
+  }
+
+  public String getMessage() {
+    return message;
+  }
+
+  public int getLineNumber() {
+    return lineNumber;
+  }
+
+  public int getStartPosition() {
+    return startPosition;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  private static String formatArgumentNumber(int argumentNumber) {
+    switch (argumentNumber){
```

Comment:
this is not very i18nable, but in any case, you can do this for all numbers by just looking at the last digit of the int and using "st" if 1, "nd" if 2, "rd" if 3 and "th" for all other values.